prover-service/src/state.rs

```diff
@@ -19,36 +20,52 @@ pub struct ProverServiceSecrets {
 }
 
 pub struct ProverServiceState {
-    pub config: ProverServiceConfig,
-    pub circuit_metadata: CircuitConfig,
+    pub prover_service_config: Arc<ProverServiceConfig>,
+    pub circuit_config: CircuitConfig,
     pub deployment_information: DeploymentInformation,
     pub groth16_vk: OnChainGroth16VerificationKey,
     pub tw_keys: TrainingWheelsKeyPair,
     pub full_prover: Mutex<FullProver>,
 }
 
 impl ProverServiceState {
-    pub fn init(deployment_information: DeploymentInformation) -> Self {
+    pub fn init(
+        prover_service_config: Arc<ProverServiceConfig>,
+        deployment_information: DeploymentInformation,
+    ) -> Self {
+        // TODO: avoid using figment and quietly merging env vars here!
```

Comment:
I'll fix this soon. Seems really scary to also load "all" environment variables 🤔